Firestore/Source/Local/FSTWriteGroupTracker.mm

```diff
@@ -40,6 +40,17 @@ - (FSTWriteGroup *)startGroupWithAction:(NSString *)action {
   return self.activeGroup;
 }
 
+- (FSTWriteGroup *)startGroupWithAction:(NSString *)action
+                         andTransaction:
+                             (firebase::firestore::local::LevelDBTransaction *)transaction {
+  // NOTE: We can relax this to allow nesting if/when we find we need it.
+  FSTAssert(!self.activeGroup,
```

Comment:
This seems to duplicate the logic already in `FSTLevelDB` that verifies no transaction is already in progress. I think you just delete `FSTWriteGroupTracker`.

Reply:
This is still in use for the memory persistence. It will be deleted when `FSTWriteGroup` is deleted.